src/NUnitEngine/nunit.engine.tests/Internal/FileSystemAccess/Default/DirectoryTests.cs

```diff
@@ -0,0 +1,205 @@
+            var expected = new SIO.DirectoryInfo(path).GetFiles().Select(x => x.FullName);
+            var directory = new Directory(path);
+
+            var actualFiles = directory.GetFiles("*");
+
+            var actual = actualFiles.Select(x => x.FullName);
+            CollectionAssert.AreEquivalent(expected, actual);
+        }
+
+        [Test]
+        public void GetFiles_NonExistingDirectory()
+        {
+            var path = SIO.Directory.GetCurrentDirectory();
+            while (SIO.Directory.Exists(path))
+            {
+                path += "a";
+            }
+            var directory = new Directory(path);
+
+            Assert.That(() => directory.GetFiles("*"), Throws.InstanceOf<SIO.DirectoryNotFoundException>());
+        }
+
+        [Test]
+        public void GetFiles_WithPattern()
+        {
+            var path = SIO.Directory.GetCurrentDirectory();
+            var expected = new SIO.DirectoryInfo(path).GetFiles("*.dll").Select(x => x.FullName);
+            var directory = new Directory(path);
+
+            var actualFiles = directory.GetFiles("*.dll");
+
+            var actual = actualFiles.Select(x => x.FullName);
+            CollectionAssert.AreEquivalent(expected, actual);
+        }
+
+        [Test]
+        public void GetFiles_SearchPatternIsNull()
+        {
+            var path = SIO.Directory.GetCurrentDirectory();
+            var directory = new Directory(path);
+
+            Assert.That(() => directory.GetFiles(null), Throws.ArgumentNullException);
+        }
+
+        [Test]
+        public void GetDirectories_NonExistingDirectory()
+        {
+            var path = SIO.Directory.GetCurrentDirectory();
+            while (SIO.Directory.Exists(path))
+            {
+                path += "a";
+            }
+            var directory = new Directory(path);
+
+            Assert.That(() => directory.GetDirectories("*", SIO.SearchOption.TopDirectoryOnly), Throws.InstanceOf<SIO.DirectoryNotFoundException>());
+        }
+
+        [Test]
+        public void GetDirectories_SearchPatternIsNull()
+        {
+            var path = SIO.Directory.GetCurrentDirectory();
+            var directory = new Directory(path);
+
+            Assert.That(() => directory.GetDirectories(null, SIO.SearchOption.TopDirectoryOnly), Throws.ArgumentNullException);
+        }
+
+        [Test]
+        public void GetDirectories_SearchOptionIsInvalid()
+        {
+            var path = SIO.Directory.GetCurrentDirectory();
+            var directory = new Directory(path);
+
+            Assert.That(() => directory.GetDirectories("*", (SIO.SearchOption)5), Throws.InstanceOf<System.ArgumentOutOfRangeException>());
+        }
+
+        private void SkipOnNonWindowsSystems()
+        {
+            if (SIO.Path.DirectorySeparatorChar != '\\')
+            {
+                Assert.Ignore("This test is not compatible with the current operating system.");
+            }
+        }
```

Comment:
How about using NUnit's PlatformAttribute instead of this method? It should just be possible to add `[Platform("Win")]` to the necessary tests - just a bit tidier, perhaps. 🙂

Reply:
I initially wanted to use the `Platform` attribute. But it is not available when the target framework is `netcoreapp1.1`.
I don't like my approach with the private method either, but I think it's better than a combination of a method to support .NET Core 1.1 and an attribute (for all other platforms).

If we want to make the exclusion more readable by using an attribute, I see two possibilities:
1. Add support for .NET Core 1.1 to `PlatformAttribute` (this would require a change in NUnit.Framework and we will need to use that version).
2.We create a new internal attribute (e.g. `WindowsOnlyAttribute`) that works for all target frameworks and contains the rudimental logic that is currently contained in `DirectoryTests.SkipOnNonWindowsSystems()`.

Reply:
Ah sorry - we've just removed the netcoreapp1.1 target recently, it must have been since you started this PR. Can you merge in master, and then you should just be able to add it in?

Reply:
Ooops, I missed the removal of netcoreapp1.1.
Replaced my OS detection with the `PlatformAttribute`. Works like a charm! 😀